src/azure-cli/azure/cli/command_modules/vm/aaz/latest/vm/disk/_attach_detach_data_disk.py

```diff
@@ -0,0 +1,517 @@
+#
+# Code generated by aaz-dev-tools
+# --------------------------------------------------------------------------------------------
+
+# pylint: skip-file
+# flake8: noqa
+
+from azure.cli.core.aaz import *
+
+
+class AttachDetachDataDisk(AAZCommand):
+    """Attach and detach data disks to/from the virtual machine.
+    """
+
+    _aaz_info = {
+        "version": "2024-03-01",
+        "resources": [
+            ["mgmt-plane", "/subscriptions/{}/resourcegroups/{}/providers/microsoft.compute/virtualmachines/{}/attachdetachdatadisks", "2024-03-01"],
+        ]
+    }
+
+    AZ_SUPPORT_NO_WAIT = True
+
+    def _handler(self, command_args):
+        super()._handler(command_args)
+        return self.build_lro_poller(self._execute_operations, self._output)
+
+    _args_schema = None
+
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        if cls._args_schema is not None:
+            return cls._args_schema
+        cls._args_schema = super()._build_arguments_schema(*args, **kwargs)
+
+        # define Arg Group ""
+
+        _args_schema = cls._args_schema
+        _args_schema.resource_group = AAZResourceGroupNameArg(
+            required=True,
+        )
+        _args_schema.vm_name = AAZStrArg(
+            options=["--vm-name"],
+            help="The name of the virtual machine.",
+            required=True,
+            id_part="name",
+        )
+
+        # define Arg Group "Parameters"
+
+        _args_schema = cls._args_schema
+        _args_schema.data_disks_to_attach = AAZListArg(
+            options=["--data-disks-to-attach"],
+            arg_group="Parameters",
+            help="The list of managed data disks to be attached.",
+        )
+        _args_schema.data_disks_to_detach = AAZListArg(
+            options=["--data-disks-to-detach"],
+            arg_group="Parameters",
+            help="The list of managed data disks to be detached.",
+        )
+
+        data_disks_to_attach = cls._args_schema.data_disks_to_attach
+        data_disks_to_attach.Element = AAZObjectArg()
+
+        _element = cls._args_schema.data_disks_to_attach.Element
+        _element.caching = AAZStrArg(
+            options=["caching"],
+            help="Specifies the caching requirements. Possible values are: **None,** **ReadOnly,** **ReadWrite.** The defaulting behavior is: **None for Standard storage. ReadOnly for Premium storage.**",
+            enum={"None": "None", "ReadOnly": "ReadOnly", "ReadWrite": "ReadWrite"},
+        )
+        _element.delete_option = AAZStrArg(
+            options=["delete-option"],
+            help="Specifies whether data disk should be deleted or detached upon VM deletion. Possible values are: **Delete.** If this value is used, the data disk is deleted when VM is deleted. **Detach.** If this value is used, the data disk is retained after VM is deleted. The default value is set to **Detach**.",
```

Comment:
```suggestion
            help="Specify whether data disk should be deleted or detached upon VM deletion. Possible values are: **Delete.** If this value is used, the data disk is deleted when VM is deleted. **Detach.** If this value is used, the data disk is retained after VM is deleted. The default value is set to **Detach**.",
```
Please use the first person voice in the help message